scripts/installer

```diff
@@ -0,0 +1,287 @@
+install() {
+  echo "Installing $TMP_FILE using kubectl ..."
+  kubectl apply -n $INSTALL_NAMESPACE -f $TMP_FILE
+  installIngress
+}
+
+# uninstall invokes kubectl delete with the downloaded manifest.
+uninstall() {
+  echo "Uninstalling $TMP_FILE using kubectl ..."
+  kubectl delete -n $INSTALL_NAMESPACE -f $TMP_FILE
+  uninstallIngress
+}
+
+# installIngress adds an ingress rule using kubectl apply.
+installIngress() {
+if [ ! -z "$INGRESS_URL" ] && [ ! -z "$INGRESS_SECRET" ]; then
+cat <<EOF | kubectl apply -n $INSTALL_NAMESPACE -f -
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: tekton-dashboard
+spec:
+  tls:
+  - hosts:
+    - $INGRESS_URL
+    secretName: $INGRESS_SECRET
+  rules:
+  - host: $INGRESS_URL
+    http:
+      paths:
+      - backend:
+          serviceName: tekton-dashboard
+          servicePort: 9097
+EOF
+elif [ ! -z "$INGRESS_URL" ]; then
+cat <<EOF | kubectl apply -n $INSTALL_NAMESPACE -f -
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: tekton-dashboard
+spec:
+  rules:
+  - host: $INGRESS_URL
+    http:
+      paths:
+      - backend:
+          serviceName: tekton-dashboard
+          servicePort: 9097
+EOF
+fi
+}
+
+# uninstallIngress removes ingress rule using kubectl delete.
+uninstallIngress() {
+  kubectl delete ingress tekton-dashboard -n $INSTALL_NAMESPACE
+}
+
+# fail_trap is executed if an error occurs.
+fail_trap() {
+  result=$?
+  cleanup
+  exit $result
+}
+
+# help provides possible cli installation arguments
+help () {
+  echo "Accepted cli arguments are:"
+  echo -e "\t[--help|-h ]\t\t\t\tPrints this help"
+  echo -e "\t[--version|-v <desired_version>]\tVersion to install, will try to fetch latest release from GitHub if not defined"
+  echo -e "\t[--openshift]\t\t\t\tWill download an openshift release"
+  echo -e "\t[--readonly]\t\t\t\tWill download a readonly release"
+  echo -e "\t[--namespace <namespace>]\t\tWill override install namespace"
+  echo -e "\t[--pipelines-namespace <namespace>]\tOverride the namespace where Tekton Pipelines is installed (defaults to tekton-pipelines)"
+  echo -e "\t[--triggers-namespace <namespace>]\tOverride the namespace where Tekton Triggers is installed (defaults to tekton-pipelines)"
+  echo -e "\t[--ingress-url <url>]\t\t\tWill create an additional ingress with the specified url"
+  echo -e "\t[--ingress-secret <secret>]\t\tWill add ssl support to the ingress"
+  echo -e "\t[--uninstall]\t\t\t\tWill uninstall instead of installing"
+}
+
+# cleanup temporary files
+cleanup() {
+  if [[ -d "${TMP_ROOT:-}" ]]; then
+    rm -rf "$TMP_ROOT"
+  fi
+}
+
+# Execution
+
+#Stop execution on any error
+trap "fail_trap" EXIT
+
+set -e
+
+# Parsing input arguments (if any)
+set -u
+
+while [[ $# -gt 0 ]]; do
+  case $1 in
+    '--version'|-v)
+       shift
+       if [[ $# -ne 0 ]]; then
+           export DESIRED_VERSION="${1}"
+       else
+           echo -e "Please provide the desired version. e.g. --version v3.0.0 or -v canary"
```

Comment:
should probably use versions we actually have here
`e.g. --version v0.7.0`

Reply:
yeah, another arg ? something like `--channel stable/nightly/dev` ?